### PR TITLE
[CIFuzz] Go support

### PR DESCRIPTION
**Pull request by Leo-Neat**
Supports projects that use go get commands rather an git clone e.g. https://github.com/google/gonids

### CODE REVIEW

File: infra/base-images/base-builder/detect_repo.py [hunk 1]
```diff
@@ -55,7 +57,14 @@ def main():
...
 
-  for single_dir in os.listdir(src_dir):
+  dirs_to_search = os.listdir(src_dir)
+  if os.path.exists(GO_PATH) and args.repo_name:
+    for root, dirs, _ in os.walk(GO_PATH):
+      for test_dir in dirs:
+        if args.repo_name in test_dir:
+          dirs_to_search.append(os.path.join(root, test_dir))
+
+  for single_dir in dirs_to_search:
```
Comment (jonathanmetzman) [inline infra/base-images/base-builder/detect_repo.py]: I'd make a `search_dirs` function, put the rest of this function in it, and replace with a call to `search_dirs`

File: infra/base-images/base-builder/detect_repo.py [hunk 2]
```diff
@@ -55,7 +57,14 @@ def main():
   else:
     src_dir = os.environ.get('SRC', '/src')
 
-  for single_dir in os.listdir(src_dir):
+  dirs_to_search = os.listdir(src_dir)
```
Comment (jonathanmetzman) [inline infra/base-images/base-builder/detect_repo.py]: I'd make a function `get_dirs_to_search` and put 60-66 in it.
I'm sure we'll have to change in the future to support other weird build systems (rust maybe).

File: infra/test_repos.py [hunk 3]
```diff
@@ -48,6 +50,7 @@
     ExampleRepo(project_name='curl',
                 oss_repo_name='curl',
                 git_repo_name='curl',
+                image_location='/src',
```
Comment (jonathanmetzman) [inline infra/test_repos.py]: Sorry, I don't quite understand why this change was needed? Can you please explain?

**Comment by jonathanmetzman** [comment]
LGTM with nits

File: infra/base-images/base-builder/detect_repo.py (hunk 2 above)
Comment (Leo-Neat) [reply infra/base-images/base-builder/detect_repo.py]: Fixed.

File: infra/test_repos.py (hunk 3 above)
Comment (Leo-Neat) [reply infra/test_repos.py]: This is required for testing. It is needed at line 93 of build_specified_commit_test.py

File: infra/base-images/base-builder/detect_repo.py (hunk 1 above)
Comment (Leo-Neat) [reply infra/base-images/base-builder/detect_repo.py]: Done.
